src/core/lib/OrderCombiner.sol

```diff
@@ -507,7 +511,10 @@ contract OrderCombiner is OrderFulfiller, FulfillmentApplier {
                         orderHash,
                         advancedOrder.numerator,
                         advancedOrder.denominator,
-                        revertOnInvalid
+                        _revertOnFailedUpdate(
```

Comment:
That's a good point about the edge case — will try adding it in (hopefully it doesn't exceed stack depth!)

I agree that it's not optimal to revert the whole batch in these cases, but in practice this will only occur when fulfilling the same order multiple times; and there's still a workaround if you have the zone itself track the remaining fill fraction using tstore

If we come up with an effective way to track fill fractions in flight then we can always roll this change back, but IMHO it's more important to guarantee that the `authorizeOrder` invariants are upheld than that orders will always be skipped if possible